tests/DCPS/DataRepresentation/DataRepresentation.cpp

```diff
@@ -22,594 +22,441 @@
-    test.test_case(explicit_default_reprs, default_repr, true);
-    test.test_case(default_repr, explicit_default_reprs, true);
-    test.test_case(xcdr2_repr, default_repr, false, false);
-    test.test_case(default_repr, xcdr2_repr, false, true, false);
-    test.test_case(xml_repr, default_repr, false, false);
-    test.test_case(default_repr, xml_repr, false, true, false);
-
-    test.set_topic(xcdr1_type.type_name(), xcdr2_repr);
-    test.test_case(default_repr, default_repr, true);
-    test.test_case(explicit_default_reprs, default_repr, true);
-    test.test_case(default_repr, explicit_default_reprs, true);
-    test.test_case(xcdr2_repr, default_repr, false, false);
-    test.test_case(default_repr, xcdr2_repr, false, true, false);
-    test.test_case(xml_repr, default_repr, false, false);
-    test.test_case(default_repr, xml_repr, false, true, false);
-
-    RegisteredType<XmlType> xml_type;
-    if (!xml_type.register_type(participant.in())) {
-      return 1;
-    }
-
-    test.set_topic(xml_type.type_name(), default_repr);
-    test.test_case(default_repr, default_repr, true, false);
-    test.test_case(explicit_default_reprs, default_repr, true, false);
-    test.test_case(default_repr, explicit_default_reprs, true, false);
-    test.test_case(xcdr2_repr, default_repr, false, false);
-    test.test_case(default_repr, xcdr2_repr, false, false);
-    test.test_case(xml_repr, default_repr, false, false, false);
-    test.test_case(default_repr, xml_repr, false, false);
-    test.test_case(xml_repr, xml_repr, true, false);
-
-    test.set_topic(xcdr1_type.type_name(), xml_repr);
-    test.test_case(default_repr, default_repr, true);
-    test.test_case(explicit_default_reprs, default_repr, true);
-    test.test_case(default_repr, explicit_default_reprs, true);
-    test.test_case(xcdr2_repr, default_repr, false, false);
-    test.test_case(default_repr, xcdr2_repr, false, true, false);
-    test.test_case(xml_repr, default_repr, false, false);
-    test.test_case(default_repr, xml_repr, false, true, false);
-    test.test_case(xml_repr, xml_repr, true, false);
-
-    if (test.any_failed()) {
-      exit_status = 1;
-    }
-
-    // Clean-up
-    DDS::ReturnCode_t rc = participant->delete_contained_entities();
-    if (rc != DDS::RETCODE_OK) {
-      ACE_ERROR((LM_ERROR, ACE_TEXT("%N:%l: main() ERROR: ")
-        ACE_TEXT("delete_contained_entities failed: %C\n"),
-        retcode_to_string(rc)));
-      exit_status = 1;
-    }
-    rc = dpf->delete_participant(participant.in());
-    if (rc != DDS::RETCODE_OK) {
-      ACE_ERROR((LM_ERROR, ACE_TEXT("%N:%l: main() ERROR: ")
-        ACE_TEXT("delete_participant failed: %C\n"),
-        retcode_to_string(rc)));
-      exit_status = 1;
-    }
-    TheServiceParticipant->shutdown();
-  } catch (const CORBA::Exception& e) {
-    e._tao_print_exception("Caught in main()");
-    exit_status = 1;
-  }
-
-  return exit_status;
+    Test test(argc, argv);
+    ret = test.run();
```

Comment:
```suggestion
    ret = test.run() ? 1 : 0;
```

It's not generally a good idea to return counts of any sort from main.